centreon/src/Core/Dashboard/Playlist/Domain/Model/NewPlaylist.php

```diff
@@ -0,0 +1,182 @@
+use Core\Dashboard\Playlist\Domain\Exception\NewPlaylistException;
+
+class NewPlaylist
+{
+    public const NAME_MIN_LENGTH = 1;
+    public const NAME_MAX_LENGTH = 255;
+    public const DESCRIPTION_MIN_LENGTH = 1;
+    public const DESCRIPTION_MAX_LENGTH = 65535;
+    public const MINIMUM_ROTATION_TIME = 10; // time in seconds
+    public const MAXIMUM_ROTATION_TIME = 60; // time in seconds
+
+    /** @var DashboardOrder[] */
+    protected array $dashboardsOrder = [];
+
+    protected ?string $description = null;
+
+    protected \DateTimeImmutable $createdAt;
+
+    /** If the author has been deleted, the author is null but the playlist is still accessible*/
+    protected ?PlaylistAuthor $author = null;
+
+    /**
+     * @param string $name
+     * @param int $rotationTime
+     * @param bool $isPublic
+     *
+     * @throws \Assert\AssertionFailedException
+     */
+    public function __construct(
+        protected string $name,
+        protected int $rotationTime,
+        protected bool $isPublic
+    ) {
+        Assertion::minLength($name, self::NAME_MIN_LENGTH, 'NewPlaylist::name');
+        Assertion::maxLength($name, self::NAME_MAX_LENGTH, 'NewPlaylist::name');
+        Assertion::range(
+            $rotationTime,
+            self::MINIMUM_ROTATION_TIME,
+            self::MAXIMUM_ROTATION_TIME,
+            'NewPlaylist::rotationTime'
+        );
+
+        $this->createdAt = new \DateTimeImmutable();
+    }
+
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    public function getRotationTime(): int
+    {
+        return $this->rotationTime;
+    }
+
+    public function isPublic(): bool
+    {
+        return $this->isPublic;
+    }
+
+    public function getCreatedAt(): \DateTimeImmutable
+    {
+        return $this->createdAt;
+    }
+
+    public function getAuthor(): ?PlaylistAuthor
+    {
+        return $this->author;
+    }
+
+    public function setAuthor(?PlaylistAuthor $author): self
+    {
+        $this->author = $author;
+
+        return $this;
+    }
+
+    /**
+     * @return DashboardOrder[]
+     */
+    public function getDashboardsOrder(): array
+    {
+        return $this->dashboardsOrder;
+    }
+
+    /**
+     * @param DashboardOrder[] $dashboardsOrder
+     *
+     * @throws NewPlaylistException
+     *
+     * @return self
+     */
+    public function setDashboardsOrder(array $dashboardsOrder): self
+    {
+        $this->dashboardsOrder = [];
+
+        foreach ($dashboardsOrder as $dashboardOrder) {
+            $this->addDashboardOrders($dashboardOrder);
+        }
+
+        return $this;
+    }
+
+    /**
+     * @param DashboardOrder $dashboardOrder
+     *
+     * @throws NewPlaylistException
+     *
+     * @return self
+     */
+    public function addDashboardOrders(DashboardOrder $dashboardOrder): self
+    {
+        $this->validateDashboardOrder($dashboardOrder);
+        $this->dashboardsOrder[] = $dashboardOrder;
```

Comment:
"addDashboardOrders" vs "$this->dashboardsOrder".
:point_right: Please check the consistency of the trailing plural "s"